src/main/navigation/navigation_multicopter.c

```diff
@@ -74,9 +74,10 @@ static void updateAltitudeVelocityController_MC(timeDelta_t deltaMicros)
 
     posControl.pids.pos[Z].output_constrained = targetVel;
 
-    // limit max vertical acceleration to 1/5G (~200 cm/s/s) if we are increasing velocity.
+    // limit max vertical acceleration to 1/5G (~200 cm/s/s) if we are increasing RoC or RoD (only if vel is of the same sign)
     // if we are decelerating - don't limit (allow better recovery from falling)
-    if (fabsf(targetVel) > fabsf(posControl.desiredState.vel.z)) {
+    const bool isSameDirection = (targetVel > 0 && posControl.desiredState.vel.z > 0) || (targetVel < 0 && posControl.desiredState.vel.z < 0);
```

Comment:
What so you think about using signbit here?

Reply:
signbit will be less practical since it returns zero for both "0" and positive numbers.

Reply:
@digitalentity If your confident that the patch will do the job.
I'm willing to give it a test, before 2.1 is released. Just to be sure.

Target - Omnibus F4 Pro